common-rest/src/test/java/org/uniprot/api/rest/controller/AbstractDownloadIT.java

```diff
@@ -0,0 +1,90 @@
+package org.uniprot.api.rest.controller;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.amqp.core.AmqpAdmin;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.RabbitMQContainer;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+import org.uniprot.api.rest.download.repository.DownloadJobRepository;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractDownloadIT extends AbstractStreamControllerIT {
+
+    @Value("${download.idFilesFolder}")
+    protected String idsFolder;
+
+    @Value("${download.resultFilesFolder}")
+    protected String resultFolder;
+
+    @Value("${async.download.queueName}")
+    protected String downloadQueue;
+
+    @Value("${async.download.retryQueueName}")
+    protected String retryQueue;
+
+    @Value(("${async.download.rejectedQueueName}"))
+    protected String rejectedQueue;
+
+    @Autowired protected AmqpAdmin amqpAdmin;
+
+    protected static RabbitMQContainer rabbitMQContainer =
+            new RabbitMQContainer(DockerImageName.parse("rabbitmq:3-management"));
+    protected static GenericContainer<?> redisContainer =
+            new GenericContainer<>(DockerImageName.parse("redis:6-alpine")).withExposedPorts(6379);
+
+    protected abstract DownloadJobRepository getDownloadJobRepository();
+
+    @DynamicPropertySource
+    public static void setUpThings(DynamicPropertyRegistry propertyRegistry) {
+        Startables.deepStart(rabbitMQContainer, redisContainer).join();
+        assertTrue(rabbitMQContainer.isRunning());
+        assertTrue(redisContainer.isRunning());
+        propertyRegistry.add("spring.amqp.rabbit.port", rabbitMQContainer::getFirstMappedPort);
+        propertyRegistry.add("spring.amqp.rabbit.host", rabbitMQContainer::getHost);
+        System.setProperty("uniprot.redis.host", redisContainer.getHost());
+        System.setProperty(
+                "uniprot.redis.port", String.valueOf(redisContainer.getFirstMappedPort()));
+        propertyRegistry.add("ALLOW_EMPTY_PASSWORD", () -> "yes");
+    }
+
+    protected void prepareDownloadFolders() throws IOException {
```

Comment:
This method prepareDownloadFolders is never used. I also noticed that we have this method duplicated for uniref(UniRefAsyncDownloadUtils) and uniprot (UniProtKBAsyncDownloadUtils). I think we have a TestUtils for async in common-rest module